### ISSUE TITLE
Pyright incorrectly reports an error when trying to reload modules

### ISSUE THREAD

**Issue by michaelkolber** [microsoft/pyright#173]
**Describe the bug**
When using `importlib` to reload a module, Pyright complains about a type mismatch, i.e. that the module is of type `Module` but it expected `ModuleType`. It is listed as an error even though it does not actually prevent compilation.

**To Reproduce**
1. Import a module.
2. Import `importlib` and try to reload a module using its `reload()` method.

**Expected behavior**
Pyright reports a **warning**. **Errors** (from my understanding) should be for code that prevents the program from running, while warnings are for suboptimal code that will nevertheless still run.

**Screenshots or Code**

**Code**
```python
import parse  # Custom module, but the problem occurs with any module
import importlib
importlib.reload(parse)  # Error happens here
```

**The error**
```
Argument of type 'Module' cannot be assigned to parameter 'module' of type 'ModuleType'
```
```json
{
	"resource": "/c:/Users/[long filepath]/file.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Argument of type 'Module' cannot be assigned to parameter 'module' of type 'ModuleType'",
	"source": "pyright",
	"startLineNumber": 12,
	"startColumn": 22,
	"endLineNumber": 12,
	"endColumn": 27
}
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? **Extension**

 Which version? **1.0.46**

**Additional context**

**No**

**Comment by erictraut** [comment]
Thanks for the bug report. This involves an easy fix which will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is fixed in 1.0.47, which I just published.

**Comment by michaelkolber** [comment]
Thanks for the quick fix!
